### DISCUSSION TITLE
Dialog Close Script, or Hide Window close 'X'

### DISCUSSION THREAD

**Discussion by robbudge** [frangoteam/FUXA#936] [Q&A] [answered]
Any way to run a script on dialog close?
or to disable the close 'x' in the top right.
i use a common dialog and tag structure in the PLC for read only information, so when the use clicks on a device the  PLC is sent an index  to copy into the standard structure.
Write tags we use the Variable bind.
When the 'Close' button is clicked a value of 255 is written to the PLC inform the PLC the common structure is not  required. This common structure can be upwards  of 30 Variables Binding them would be a nightmare for 50 devices.

The problem arrises when the user simply clicks the 'X' and the dialog closes. the PLC is still copying  the data and the common area is  utilized.

a simple option would be  make the 'X' White so still active just not visible, Any ideas ???

**Comment by unocelli** [comment]
Hi,
I understand the problem, i think a dialog should always have a cancel (and that was the idea of the 'X'). I think the simplest solution is to give the option to hide the 'X'. currently this option is not available.

**Comment by robbudge** [comment, marked as answer]
I did find a work around,   in the client file styles.12ccc0f8044a081d.css 
I modified the color of the close 'X' from grey to White.
.material-icons{color:rgba(255, 255, 255, 0.9);
The Button  is still active, but with the white dialog box, not visible.
Now the  operator automatically selects the close button.
I have been forced to use this approach in the past
![image](https://github.com/frangoteam/FUXA/assets/15365138/629e2f87-fbb5-4181-b3c1-615949a128a6)
